src/Validation/URLValidationCron.php

```diff
@@ -0,0 +1,116 @@
+<?php
+/**
+ * WP cron process to validate URLs in the background.
+ *
+ * @package AMP
+ * @since 2.1
+ */
+
+namespace AmpProject\AmpWP\Validation;
+
+use AmpProject\AmpWP\BackgroundTask\CronBasedBackgroundTask;
+
+/**
+ * URLValidationCron class.
+ *
+ * @since 2.1
+ *
+ * @internal
+ */
+final class URLValidationCron extends CronBasedBackgroundTask {
+	/**
+	 * The cron action name.
+	 *
+	 * @var string
+	 */
+	const BACKGROUND_TASK_NAME = 'amp_validate_urls';
+
+	/**
+	 * The number of URLs to check per type each time the cron action runs.
+	 *
+	 * @var int
+	 */
+	const DEFAULT_LIMIT_PER_TYPE = 1;
+
+	/**
+	 * The length of time, in seconds, to sleep between each URL validation.
+	 *
+	 * @var int
+	 */
+	const DEFAULT_SLEEP_TIME = 1;
+
+	/**
+	 * Callback for the cron action.
+	 */
+	public function process() {
+		$number_per_type         = $this->get_url_validation_number_per_type();
+		$validation_url_provider = new ScannableURLProvider( $number_per_type, [], true );
+		$urls                    = $validation_url_provider->get_urls();
+		$sleep_time              = $this->get_sleep_time();
+
+		$validation_provider = new URLValidationProvider();
+
+		$validation_provider->with_lock(
```

Comment:
@westonruter Here's the next use case for `with_lock` outside the CLI script. What do you think? Is it fine if, in some rare scenarios, this cron task validates a bunch of URLs at the same time someone is running the CLI script? If so, I'll remove the locking. 

My thinking was that having multiple validation processes running simultaneously could overload some servers. But on the other hand, by default we're only validating a few URLs at a time, and sleeping between each requests. Anyone with the knowledge to validate a lot of URLs at a time can do so at their own risk.

Reply:
I think that it is going to be very unlikely that people will be often using the CLI script to do validation. Most likely the vast majority of sites that have limited resources are shared servers on which hosts don't even offer WP-CLI access in the first place. So I think that the locking isn't necessary right now, but it's something we can keep in mind in the future if need be.

Reply:
Sounds good. I'll update.

Reply:
Removed locking in 2d2a0a62b